src/rules/noVarKeywordRule.ts

```diff
@@ -37,46 +38,50 @@ export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "Forbidden 'var' keyword, use 'let' or 'const' instead";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        const noVarKeywordWalker = new NoVarKeywordWalker(sourceFile, this.getOptions());
-        return this.applyWithWalker(noVarKeywordWalker);
+        return this.applyWithFunction(sourceFile, walk);
     }
 }
 
-class NoVarKeywordWalker extends Lint.RuleWalker {
-    public visitVariableStatement(node: ts.VariableStatement) {
-        if (!Lint.hasModifier(node.modifiers, ts.SyntaxKind.DeclareKeyword)
-                && !Lint.isBlockScopedVariable(node)) {
-            this.reportFailure(node.declarationList);
-        }
-
-        super.visitVariableStatement(node);
-    }
+function walk(ctx: Lint.WalkContext<void>): void {
+    const { sourceFile } = ctx;
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+        switch (node.kind) {
+            case ts.SyntaxKind.VariableStatement: {
+                const vs = node as ts.VariableStatement;
+                if (!Lint.isBlockScopedVariableDeclarationList(vs.declarationList) && !isGlobalVarDeclaration(vs)) {
+                    fail(vs.declarationList);
+                }
+                break;
+            }
 
-    public visitForStatement(node: ts.ForStatement) {
-        this.handleInitializerNode(node.initializer);
-        super.visitForStatement(node);
-    }
+            case ts.SyntaxKind.ForStatement:
+            case ts.SyntaxKind.ForInStatement:
+            case ts.SyntaxKind.ForOfStatement: {
+                const { initializer } = node as ts.ForStatement | ts.ForInStatement | ts.ForOfStatement;
+                if (initializer && isVariableDeclarationList(initializer) && !Lint.isBlockScopedVariableDeclarationList(initializer)) {
+                    fail(initializer);
+                }
+                break;
+            }
+        }
 
-    public visitForInStatement(node: ts.ForInStatement) {
-        this.handleInitializerNode(node.initializer);
-        super.visitForInStatement(node);
-    }
+        return ts.forEachChild(node, cb);
+    });
 
-    public visitForOfStatement(node: ts.ForOfStatement) {
-        this.handleInitializerNode(node.initializer);
-        super.visitForOfStatement(node);
+    function fail(node: ts.Node): void {
+        // Don't apply fix in a declaration file, because may have meant 'const'.
+        const fix = sourceFile.isDeclarationFile ? undefined : new Lint.Replacement(node.getStart(), "var".length, "let");
+        ctx.addFailureAtNode(Lint.childOfKind(node, ts.SyntaxKind.VarKeyword)!, Rule.FAILURE_STRING, fix);
```

Comment:
there is no need for `addFailureAtNode` with `childOfKind`, you already computed the position of the var keyword for the fix one line above.